core/src/Config.purs

```diff
@@ -481,7 +483,21 @@ readConfig :: forall a. FilePath -> Spago (LogEnv a) (Either String { doc :: Yam
 readConfig path = do
   logDebug $ "Reading config from " <> path
   FS.exists path >>= case _ of
-    false -> pure $ Left $ case path of
-      "spago.yaml" -> "Did not find " <> path <> " Run `spago init` to initialise a new project."
-      _ -> "Did not find " <> path
+    false -> do
+      yml <- case map (_ <> ".yml") $ String.stripSuffix (String.Pattern ".yaml") path of
+        Nothing -> pure Nothing
+        Just y -> do
+          hasYml <- FS.exists y
+          pure $
+            if hasYml then
+              Just y
+            else
+              Nothing
```

Comment:
```suggestion
            pure $ guard hasYml *> pure y
```

Reply:
I find `guard` harder to read than the if/else above, let's keep that one